### DISCUSSION TITLE
Ambiente de homologação.

### DISCUSSION THREAD

**Discussion by brunoffreire** [TadaSoftware/PyNFe#270] [General]
Boa tarde.

Para desenvolver um sistema usando esta lib e testar a emissão de NF em ambiente de homologação, é necessário um certificado digital verdadeiro?

É possível desenvolver o software e testar com um certificado auto-assinado... ou obter um certificado de testes?

Alguém sabe orientar sobre esse tópico?

Obrigado.

**Comment by juniortada** [comment]
Boa noite @brunoffreire 
Sim, para testar emissão em ambiente de homologação é necessário um certificado digital A1 válido.
O que você pode fazer sem o certificado digital é a serialização do xml para emissão de notas, mas não será possível assinar nem enviar este xml.

**Reply by brunoffreire** [reply]
Entendido. Obrigado pela resposta.

Outra pergunta. Se o certificado digital for de uma empresa sem inscrição estadual (por que só presta serviços, não paga ICMS)... Esse certificado pode ser usado para teste de emissão de nota de venda de produtos no ambiente de homologação da SEFAZ?

Também li na internet que poderia ser usado e-cpf no ambiente de homologação da SEFAZ. Alguém sabe?

**Reply by juniortada** [reply]
Pode ser usado sim. Dependendo do autorizador (pode variar a regra dependendo da UF) precisa somente de cadastro para autorizar emissão.
